tests/unit/client.py

```diff
@@ -262,28 +268,69 @@ def avoid_network_race_condition():
     for timeout in range(10):
         que = db.session.query(NetworkingAction).count()
         if (que > 0):
-            time.sleep(1)
+            time.sleep(.5)
         else:
             return
     raise TimeoutError(" Timed out to avoid race condition. ")
 
 
+def port_has_server(url):
+    """ Returns whether a local TCP port is a connectable http server. """
+
+    try:
+        sess = requests.Session()
+        sess.auth = (username, password)
+        sess.get(url + '/projects')
+        return True
+    except:
+        return False
+
+
+def wait_for_service(url, timeout=15):
+    """
+    Waits for a port to become a working http server
+    timeout -- number of seconds to wait (default 60)
+    """
+
+    begin = time.time()
+    while True:
+        try:
+            sess = requests.Session()
+            sess.auth = (username, password)
+            sess.get(url + '/projects')
+            break
```

Comment:
I wanted to catch **only** `ConnectionError` so as not to mask other potential exceptions coming from requests. port_has_server() catches everything.